okio-resourcefilesystem/src/main/kotlin/okio/resourcefilesystem/ResourceFileSystem.kt

```diff
@@ -57,10 +57,21 @@ class ResourceFileSystem internal constructor(
     return fileSystem.list(fileSystemPath).filterNot { it.name.endsWith(".class") }
   }
 
-  override fun open(file: Path): FileHandle {
+  override fun open(
+    file: Path,
+    read: Boolean,
+    write: Boolean
+  ): FileHandle {
     val (fileSystem, fileSystemPath) = toSystemPath(file)
       ?: throw FileNotFoundException("file not found: $file")
-    return fileSystem.open(fileSystemPath)
+    return fileSystem.open(
+      file = fileSystemPath,
+      read = read,
+      write = write,
+      createIfAbsent = createIfAbsent,
```

Comment:
How does this work?

Reply:
It’s in StandardOpenOptions.
https://docs.oracle.com/javase/7/docs/api/java/nio/file/StandardOpenOption.html#CREATE

(I’ve since changed my mind on exposing this, not completely decided either way).